providers/root-provider.tsx

```diff
@@ -12,7 +13,9 @@ export function RootProvider ({
     <ReactQueryProvider>
       <CosmosChainProvider>
         <NextIntlClientProvider messages={messages}>
-          {children}
+          <ModalProvider>
```

Comment:
We have already too much providers here. I think we can create some global providers collector component

Something like this
https://egor-xyz.medium.com/how-to-combine-providers-in-react-e553a155d0b9